t/03-testapi.t

```diff
@@ -334,6 +334,21 @@ subtest 'script_run' => sub {
     is(background_script_run('sleep 10', output => 'foo'), '1234', 'background_script_run with output returns valid PID');
 };
 
+sub assert_script_sudo_test ($is_serial_terminal) {
+    my $mock_testapi = Test::MockModule->new('testapi');
+    $mock_testapi->redefine(_handle_found_needle => sub { return $_[0] });
+    $mock_testapi->noop(qw(send_key enter_cmd));
+    $mock_testapi->redefine(hashed_string => 'XXX');
+    $mock_testapi->redefine(wait_serial => "XXX-0-");
+    my $info = $is_serial_terminal ? 'with serial_terminal' : 'without serial_terminal';
+    is(assert_script_sudo('echo foo'), undef, 'sucessfull assertion of script_sudo $info');
```

Comment:
If someone changes the `assert_script_sudo` to what it did before, then the test would still pass, so I would say that the test is not sufficient.
Maybe you can mock `script_sudo` and check the parameter in the mocked function.

Also `s/successfull/successful/` :)

Reply:
@b10n1k I see you added mocking. Please don't forget about the typo.

Reply:
*this should be ready for review*